src/concepts/rendering.md

```diff
@@ -77,3 +77,231 @@ This 4 minute talk about `IMGUI` is also tangentially relevant.
 - **Architecture design considerations**: With `ratatui`, out of the box, there's little to no help
   in organizing large applications. Ultimately, the decision on structure and discipline rests with
   the developer to be principled.
+
+## How does Ratatui work?
+
+The smallest unit of data in Ratatui is a [`Cell`].
+
+```rust
+pub struct Cell {
+  pub symbol: String,
+  pub fg: Color,
+  pub bg: Color,
+  #[cfg(feature = "crossterm")]
+  pub underline_color: Color,
+  pub modifier: Modifier,
+  pub skip: bool,
+}
+```
+
+You can see that a `Cell` has a `String` which is the symbol that will be drawn in that `Cell` as
+well as `Style` information (foreground color, background color, modifiers etc).
+
+And a [`Buffer`] is a collection of `Cell`s:
```

Comment:
Buffer is more important than Cell as that's what the widgets interact with, so perhaps cover it first and then go down a level.